spark/src/main/java/org/apache/iceberg/spark/source/IcebergBatchWriter.java

```diff
@@ -78,31 +81,31 @@
 import static org.apache.iceberg.TableProperties.WRITE_TARGET_FILE_SIZE_BYTES;
 import static org.apache.iceberg.TableProperties.WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT;
 
-// TODO: parameterize DataSourceWriter with subclass of WriterCommitMessage
-class Writer implements DataSourceWriter {
-  private static final Logger LOG = LoggerFactory.getLogger(Writer.class);
+class IcebergBatchWriter implements BatchWrite {
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergBatchWriter.class);
 
   private final Table table;
   private final FileFormat format;
   private final FileIO fileIo;
   private final EncryptionManager encryptionManager;
-  private final boolean replacePartitions;
+  private final TableCapability writeBehavior;
```

Comment:
This is used to communicate capabilities to Spark. I don't think that Iceberg should misuse it. Instead, Iceberg should keep track of the configuration set in the write builder.